ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -190,27 +190,217 @@ define(['QUnit', 'jquery', 'manage_taxonomies', 'react',
   QUnit.test('Assert that TermComponent renders properly',
     function(assert) {
       assert.ok(TermComponent, "class object not found");
+      var done = assert.async();
       var term = {
         "id": 9,
         "slug": "test",
         "label": "test",
         "weight": 1
       };
-      var termComponentRendered = React.addons.TestUtils.
+      var parentUpdateCount = 0;
+      var updateTerm = function() {
+        parentUpdateCount += 1;
+      };
+      TestUtils.initMockjax({
+        url: "/api/v1/repositories/repo/vocabularies/difficulty/terms/test",
+        responseText: term,
+        type: "PUT"
+      });
+      var afterMount = function(component) {
+        var labels = React.addons.TestUtils.
+        scryRenderedDOMComponentsWithTag(
+          component,
+          'label'
+        );
+        //testing label value render
+        var termLable = labels[0];
+        var label = termLable.getDOMNode();
+        assert.equal(label.innerHTML, term.label);
+        assert.equal(component.state.formatActionState, 'edit');
+        var formatButton = React.addons.TestUtils.
```

Comment:
Please add some line breaks to break the test into logical pieces for easier readability